src/offchainapi/protocol.py

```diff
@@ -154,47 +152,32 @@ def __init__(self, myself, other, vasp, storage, processor):
                 self.other_address_str
             )
 
-        # A reentrant lock to manage access.
-        self.rlock = RLock()
-
-        # State that is persisted.
+        root = self.storage.make_dir(self.myself.as_str())
+        other_vasp = self.storage.make_dir(
+            self.other_address_str, root=root
+        )
 
-        root = self.storage.make_value(self.myself.as_str(), None)
-        other_vasp = self.storage.make_value(
-            self.other_address_str, None, root=root
+        # The map of commited requests with their corresponding responses.
+        # This is also used to ensure command responses are indempotent.
+        # All requests in this store have response attached.
+        self.command_sequence = self.storage.make_dict(
```

Comment:
The name `sequence` is really a misnomer here. What about `committed_commands` or `command_set`?

Reply:
makes sense! renamed to `committed_commands`